tests/conftest.py

```diff
@@ -204,3 +205,99 @@ def _R(expr):
         assert len(base) == 1
         base = base.pop()
     return EvalDerivative(*expr.args, base=base)
+
+
+# Utilities for testing tree structure
+
+
+def assert_structure(operator, exp_trees=None, exp_iters=None):
+    """
+    Utility function that helps to check loop structure of IETs. Retrieves trees from an
+    Operator and check that blocking structure is as expected. Trees and Iterations are
+    returned for further use in tests.
+
+    Example:
+
+    To check the following structure:
+
+    .. code-block:: python
+
+        for time
+            for x
+                for y
+            for f
+                for y
+
+    we call:
+
+    `trees, iters = assert_structure(op, ['t,x,y', 't,f,y'], 't,x,y,f,y')`
+    """
+
+    mapper = {'time': 't'}
+    trees = retrieve_iteration_tree(operator)
+    iters = FindNodes(Iteration).visit(operator)
+
+    if exp_trees is not None:
+        exp_trees = [i.replace(',', '') for i in exp_trees]  # 't,x,y' -> 'txy'
+        tree_struc = (["".join(mapper.get(i.dim.name, i.dim.name) for i in j)
+                       for j in trees])
+        assert tree_struc == exp_trees
+
+    if exp_iters is not None:
+        exp_iters = exp_iters.replace(',', '')  # 't,x,y' -> 'txy'
+        iter_struc = "".join(mapper.get(i.dim.name, i.dim.name) for i in iters)
+        assert iter_struc == exp_iters
+
+    return trees, iters
+
+
+def get_blocked_nests(operator, exp_nests):
+    """
+    Utility function that helps to check existence of blocked nests. The structure of the
+    operator is not used. Trees are returned for further use in tests.
+
+    Example:
+
+    For the following structure:
+
+    .. code-block:: python
+
+        for time
+            for x0_blk0
+                for x
+            for x1_blk0
+                for x
+
+    we call:
+
+    `trees, bns = get_blocked_nests(op, {'x0_blk0', 'x1_blk0'})`
+
+    to assert the existence of 'x0_blk0', 'x1_blk0' and then the function returns the
+    operator trees and a dictionary with the first blocking Iterations that start a
+    blocking subtree
+
+    """
+    bns = {}
+    iterations = []
+    trees = retrieve_iteration_tree(operator)
+    for tree in trees:
+        iterations = [i for i in tree if i.dim.is_Incr]  # Collect Incr dimensions
+        if len(iterations):
+            # If Incr dimensions exist map the first one to its name in the dict
+            bns[iterations[0].dim.name] = iterations[0]
```

Comment:
so if there were **two** `x0_blk0` , then only one would be tracked, right? 
so this is based on the assumption that there's only one at most -- ensured by construction?

Reply:
I mean, two **different** `x0_blk0` clearly (not shared ones by two or more subtrees, which are fine to override, as they'd be mapped to the very same `iterations[0]`)

Reply:
Wait really no chance? Like what happens if you have two stencil with dependencies that force two `x0_blk` is it guaranteed to be a different variable?

Reply:
Every new blocking dimension gets a new name (different variable). Is there something specific you have in mind( as in pseudocode)

Reply:
> is it guaranteed to be a different variable?

I think  so, by construction, and that makes sense to me, as the two loop nests may have different optimal block shape